### PR TITLE
bug 1647364: Switch default branch from master to main

### PR DESCRIPTION
**Pull request by jwhitlock**
Update the configuration and documentation for today's change of making the new `main` branch the default branch (see [this blog post](https://www.hanselman.com/blog/EasilyRenameYourGitDefaultBranchFromMasterToMain.aspx), as well as some other updates:

* In CircleCI, take special actions like uploading docker images on the `main` rather than `master` branch
* Update `bin/release.py` to latest version that supports different default branches, plus a line length change
* Update documentation references and code comments to the `main` branch of Ichnaea
* Bump copyright to 2020
* Fix some URLs and surrounding docs that failed `make linkcheck` validation

Addition, non-code changes are also needed, see [bug 1647364](https://bugzilla.mozilla.org/show_bug.cgi?id=1647364) for tracking.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -32,5 +35,4 @@ Legal
 
 Currently we don't have any legal contributor agreement, so code
 ownership stays with the original authors. The project is licensed
-under the `Apache License Version 2
-<https://github.com/mozilla/ichnaea/blob/master/LICENSE>`_.
+under the `Apache License Version 2 <./LICENSE>`_.
```
Comment (willkg) [inline CONTRIBUTING.rst:38]: I think this link won't work in readthedocs.

File: bin/release.py [hunk 2]
```diff
@@ -213,10 +215,22 @@ def run():
...
+
+    # Let's make sure we're up-to-date and on main branch
     current_branch = check_output("git rev-parse --abbrev-ref HEAD")
-    if current_branch != "master":
-        print(f"Must be on the master branch to do this (not {current_branch})")
+    if current_branch != main_branch:
+        print(
+            f"Must be on the {main_branch} branch to do this;"
+            f" currently on {current_branch}"
+        )
```
Comment (willkg) [inline bin/release.py:233]: If you fix the flake8 configuration to ignore E501, then you don't have to do this.

File: setup.cfg [hunk 3]
```diff
@@ -27,6 +27,7 @@ github_user = mozilla
 github_project = ichnaea
 bugzilla_product = Location
 bugzilla_component = General
+main_branch = main
 
 [tool:paul-mclendahand]
 github_user = mozilla
```
Comment (willkg) [inline setup.cfg:33]: You're going to need to add `main_branch` for paul-mclendahand, too.

File: CONTRIBUTING.rst (hunk 1 above)
Comment (jwhitlock) [reply CONTRIBUTING.rst:38]: This file isn't included in docs, but I'll return to the full version.

**Comment by jwhitlock** [comment]
Force-push to address review feedback:
* Use full URL rather than relative URL for Apache license link
* Ignore `E501`, line too long, in flake8
* Add `main-branch` for [paul-mclendahand](https://github.com/willkg/paul-mclendahand) as well

File: CONTRIBUTING.rst (hunk 1 above)
Comment (willkg) [reply CONTRIBUTING.rst]: Oh, screwy. I usually include it in the docs and I must have missed this project. For example:

https://github.com/mozilla-services/socorro/blob/master/docs/contributing.rst

**Comment by jwhitlock** [comment]
Thanks!

**Pull request merged**